mh/state.py

```diff
@@ -589,9 +792,518 @@ def match_city(city, fields):
                 ])
         return cities
 
+    def update_players(self):
+        # Central server method for clearing unused Capcom IDs
+        for capcom_id, player_info in self.capcom_ids.items():
+            if player_info["session"] is not None and \
+                player_info["session"].local_info["server_id"] and \
+                capcom_id not in self.cache.players:
+                self.capcom_ids[capcom_id]["session"] = None
+            elif capcom_id in self.cache.players:
+                self.capcom_ids[capcom_id]["session"] = self.cache.players[capcom_id]
+
+    def update_capcom_id(self, session):
+        # Central server method for keeping track of in-use Capcom IDs
+        self.capcom_ids[session.capcom_id]["session"] = session
+
 
 CURRENT_STATE = State()
 
 
 def get_instance():
     return CURRENT_STATE
+
+
+class PacketTypes:
+    FriendlyHello = 0x0001
+    ReqConnectionInfo = 0x0002
+    SendConnectionInfo = 0x0003
+    ReqServerRefresh = 0x0004
+    SessionInfo = 0x0005
+
+
+class CentralConnectionHandler(object):
```

Comment:
Ditto, it has to be in its own file. Do you have a solid reason not to inherit from an existing python class?

Reply:
Moved to `other/cache.py`